.gitlab-ci.yml

```diff
@@ -6,329 +6,167 @@ stages:
-  extends: .test
+  stage: prepare
+  extends: [.ascgit_job, .darwin_job]
   variables:
-    <<: *a100
-    <<: *buildtype_release
-    <<: *gpu
+    GIT_SUBMODULE_STRATEGY: none
+  script:
+   - .gitlab/download_prereq.sh
+  artifacts:
+    paths:
+      - goldfiles.tar.gz
 
-install_gnu_skylake_fort:
+openmpi_gcc:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
+    SPACK_ENV_NAME: openmpi-gcc
 
-install_gnu_skylake_fort_kokkos:
+openmpi_fortran_gcc:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
-    SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+python+mpi+tests+kokkos+spiner+hdf5+kokkos-kernels%gcc@=${SINGULARITY_EOS_GCC_VERSION} ^kokkos std=${SINGULARITY_EOS_CXX_STANDARD} +openmp ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
+    SPACK_ENV_NAME: openmpi-fortran-gcc
 
-install_gnu_skylake_fort_kokkos_cxx17:
+openmpi_cuda_gcc_ampere:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
-    SINGULARITY_EOS_CXX_STANDARD: 17
-    SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+python+mpi+tests+kokkos+spiner+hdf5+kokkos-kernels%gcc@=${SINGULARITY_EOS_GCC_VERSION} ^kokkos std=${SINGULARITY_EOS_CXX_STANDARD} +openmp ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
+    SPACK_ENV_NAME: openmpi-cuda-gcc-ampere
+    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p shared-gpu-ampere"
 
-install_gnu_skylake_fort_static_kokkos:
+openmpi_cuda_gcc_volta:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
-    SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+python+mpi+tests+kokkos+spiner+hdf5+kokkos-kernels%gcc@=${SINGULARITY_EOS_GCC_VERSION} ^kokkos std=${SINGULARITY_EOS_CXX_STANDARD} ~shared +openmp ^kokkos-kernels~shared ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
-
-install_gnu_skylake_nofort:
+    SPACK_ENV_NAME: openmpi-cuda-gcc-volta
+    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p volta-x86 -C cpu_family:haswell"
+
+# currently disabled due to very long compilation times
+#openmpi_fortran_cuda_gcc_ampere:
+#  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
+#  needs: [download_prereq]
+#  variables:
+#    SPACK_ENV_NAME: openmpi-fortran-cuda-gcc-ampere
+#    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p shared-gpu-ampere"
+#
+#openmpi_fortran_cuda_gcc_volta:
+#  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
+#  needs: [download_prereq]
+#  variables:
+#    SPACK_ENV_NAME: openmpi-fortran-cuda-gcc-volta
+#    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p volta-x86 -C cpu_family:haswell"
```

Comment:
As discussed offline---after this is merged in, maybe next FY when we have human-cycles, we need to find a way to get these running in reasonable amounts of time.